File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        if k==1:
            return n
    
        arr = []

        for x in range(n):
            arr.append(x+1)
    
        while len(arr) >= k:
            arr = arr[k:] + arr[:k-1]
        # print(arr)
            
        while len(arr) > 1:
            d = k % len(arr)

            if d==0:
                arr = arr[:-1]
            else:
                arr = arr[d:] + arr[:d-1]

        

        return arr[0]

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number}
 */
var findTheWinner = function(n, k) {
    
    // 20
    // 6
    // [ 9, 13, 16, 20, 1 ]
    // 9 13 16 20 1 
    // 9 13 16 20 1 
    // 13 16 20 1
    // 20 1 13 
    // 20 1 
    // 20
    
    //3
    // 1
    // [1,2,3]
    // 1 2 3
    // 2 3
    // 3
    if(k===1) return n
    
    let arr = []
    for(let i=1;i<=n;i++)
        arr.push(i)
    
    while(arr.length >= k ){
        arr=[...arr.slice(k), ...arr.slice(0,k-1)]
        
    }
    
    while(arr.length>1){
        let d = k%arr.length
        
        if(d===0)
            arr = arr.slice(0,-1)
        else
            arr=[...arr.slice(d), ...arr.slice(0,d-1)]
        
    }
    // console.log(arr)
    return arr[0]
};